runtime/src/main/cpp/Cleaner.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+#include "Cleaner.h"
+
+#include "Memory.h"
+#include "Runtime.h"
+#include "Worker.h"
+
+// Defined in Cleaner.kt
+extern "C" void Kotlin_CleanerImpl_shutdownCleanerWorker(bool);
+
+namespace {
+
+struct CleanerImpl {
+  ObjHeader header;
+  KInt worker;
+  KRef clean;
+};
+
+bool cleanersDisabled = false;
+
+void disposeCleaner(CleanerImpl* thiz) {
+    if (atomicGet(&cleanersDisabled)) {
+        if (Kotlin_cleanersLeakCheckerEnabled()) {
+            konan::consoleErrorf(
+                    "Cleaner %p was disposed during program exit\n"
+                    "Use `Platform.isCleanersLeakCheckerActive = false` to avoid this check.\n",
+                    thiz);
+            RuntimeCheck(false, "Terminating now");
+        }
+        return;
+    }
+
+    WorkerSchedule(thiz->worker, thiz->clean);
```

Comment:
`WorkerSchedule` seems to use `CreateStablePointer` (~`addHeapRef`) internally.
We better make absolutely sure this is safe during (that phase of) GC.

Reply:
Alternatively we could make a buffer that gets processed at the end of GC, where it should be more safe (at least it is similar to `incrementStack` at the beginning of GC).

Reply:
Fixed by making `WorkerSchedule` accept a stable pointer to a job. Stable pointer is made at the `createCleaner` and just passed to `WorkerSchedule` during GC.